DGWebApi/Controllers/AgifyController.cs

```diff
@@ -0,0 +1,51 @@
+﻿using DGWebApi.Services.Interfaces;
+using DGWebApi.Utils;
+using Microsoft.AspNetCore.Mvc;
+
+// For more information on enabling Web API for empty projects, visit https://go.microsoft.com/fwlink/?LinkID=397860
+
+namespace DGWebApi.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class AgifyController : ControllerBase
+    {
+        [HttpGet("{name}")]
+        public async Task<IActionResult> GetByName(string name, IAgifyService agifyService)
+        {
+            var nameCount = await agifyService.GetName(name);
+
+            if (nameCount == null)
+            {
+                return StatusCode(500);
```

Comment:
You don't want to intentionally return 500 errors. 500 means an unexpected error so it shouldn't be returned as normal flow

Reply:
I've made it 404.